rll/rightscale-mirrors.sh

```diff
@@ -0,0 +1,300 @@
+#! /bin/bash -e
+
+# RightScale provides mirrors of some OS distributions. Snapshots of these mirrors
+# are taken daily so that any the mirrors can be "frozen" to any given day. These
+# mirrors also usually come from a fixed IP range for firewall friendliness. The
+# mirrors may be browsed at mirror.rightscale.com. For further information, see
+# https://support.rightscale.com/12-Guides/RightScale_101/System_Architecture/Rightscale_OS_Software_Mirrors/.
+# The following are available:
+#   Ubuntu main, universe, security-updates mirrors (/ubuntu_daily)
+#   CentOS Base, addons, extras, update mirrors (/centos)
+#   Fedora EPEL (/epel)
+#   Rubygems (/rubygems)
+
+# Mirrors are currently only supported for the following distros:
+#   Ubuntu 12.04/14.04, CentOS 6/7
+# Additionally, on Redhat 6/7 Fedora EPEL will be set up, but not the main repository mirrors.
+
+# To enable this functionality, have this script be the FIRST script in your
+# boot sequence. The following parameters are accepted:
+#   FREEZE_DATE - The date from which to pull OS/EPEL software. Accepts following parameters:
+#       empty string - disables usage of frozen mirrors (use config already on disk)
+#       YYYY-MM-DD - i.e 2015-06-01
+#       latest - today's date
+#   RUBYGEMS_FREEZE_DATE -  Date from which to pull Rubygems software. Same parameters as above.
+#   MIRROR_HOST - host, possible values:
+#     ENV:RS_ISLAND - Island load balancers for your RightScale account. Note that this
+#                     is NOT guaranteed to be geographically closest, but is rather
+#                     assigned per acocunt
+#     cf-mirror.rightscale.com - CloudFront version of mirror.rightscale.com
+
+FREEZE_DATE=${FREEZE_DATE//-/}
+if [[ -z "$FREEZE_DATE" ]]; then
+  echo "Since FREEZE_DATE is not set, not configuring RightScale software mirrors"
+elif [[ ! "$FREEZE_DATE" =~ ^[0-9]{8}$ ]] && [[ ! "$FREEZE_DATE" = "latest" ]]; then
+  echo "If FREEZE_DATE is set, it must be either 'latest' or in the format YYYY-MM-DD"
+  exit 1
+fi
+today=$(date +"%Y%m%d")
+if [[ "$FREEZE_DATE" =~ ^[0-9]{8}$ ]] && [[ $FREEZE_DATE -gt $today ]]; then
+  echo "FREEZE_DATE can't be from a date in the future"
+  exit 1
+fi
+
+RUBYGEMS_FREEZE_DATE=${RUBYGEMS_FREEZE_DATE//-/}
+if [[ -z "$RUBYGEMS_FREEZE_DATE" ]]; then
+  echo "Since RUBYGEMS_FREEZE_DATE is not set, not configuring Rubygems mirror"
+elif [[ ! "$RUBYGEMS_FREEZE_DATE" =~ ^[0-9]{8}$ ]] && [[ ! "$RUBYGEMS_FREEZE_DATE" = "latest" ]]; then
+  echo "If RUBYGEMS_FREEZE_DATE is set, it must be either 'latest' or in the format YYYY-MM-DD"
+  exit 1
+fi
+today=$(date +"%Y%m%d")
```

Comment:
`today` is already set on line 38.  Can we set it in a single place, perhaps towards the top.